host/src/FunctionsNetHost/Program.cs

```diff
@@ -35,25 +34,36 @@ private static async Task<GrpcWorkerStartupOptions> GetStartupOptionsFromCmdLine
             var workerOption = new Option<string>("--workerId");
             var grpcMsgLengthOption = new Option<int>("--grpcMaxMessageLength");
             var requestIdOption = new Option<string>("--requestId");
+            var funcUriOption = new Option<string>("--functions-uri");
+            var funcRequestIdOption = new Option<string>("--functions-request-id");
+            var funcWorkerIdOption = new Option<string>("--functions-worker-id");
+            var funcGrpcMsgLengthOption = new Option<int?>("--functions-grpc-max-message-length");
 
-            var rootCommand = new RootCommand();
+            var rootCommand = new RootCommand
+            {
+                TreatUnmatchedTokensAsErrors = false
+            };
             rootCommand.AddOption(portOption);
             rootCommand.AddOption(hostOption);
             rootCommand.AddOption(workerOption);
             rootCommand.AddOption(grpcMsgLengthOption);
             rootCommand.AddOption(requestIdOption);
+            rootCommand.AddOption(funcUriOption);
+            rootCommand.AddOption(funcRequestIdOption);
+            rootCommand.AddOption(funcWorkerIdOption);
+            rootCommand.AddOption(funcGrpcMsgLengthOption);
 
             var workerStartupOptions = new GrpcWorkerStartupOptions();
 
-            rootCommand.SetHandler((host, port, workerId, grpcMsgLength, requestId) =>
+            rootCommand.SetHandler((context) =>
             {
-                workerStartupOptions.Host = host;
-                workerStartupOptions.Port = port;
-                workerStartupOptions.WorkerId = workerId;
-                workerStartupOptions.GrpcMaxMessageLength = grpcMsgLength;
-                workerStartupOptions.RequestId = requestId;
-            },
-                hostOption, portOption, workerOption, grpcMsgLengthOption, requestIdOption);
+                var serverEndpoint = context.ParseResult.GetValueForOption(funcUriOption) ?? $"http://{context.ParseResult.GetValueForOption(hostOption)}:{context.ParseResult.GetValueForOption(portOption)}";
```

Comment:
Synced with Jon from the .net command line team and he suggested I should switch to the new approach which uses the `InvocationContext` and calling the `GetValueForOption` method. This is something they have introduced in the newer version.

> This is closer to the pattern we're arriving at in the next version. Handler parameter binding is going away as a concept.